src/components/Fuel/FuelForm.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { Button, Label, Select, TextInput } from 'flowbite-react';
+import {
+  FuelSourceUnit,
+  FuelSourceType,
+  UnregisteredFuelRequest,
+} from '../../schema/fuel.schema';
+
+interface FuelFormProps {
+  handleSubmit: (fuelData: UnregisteredFuelRequest) => void;
+}
+
+export default function FuelForm({ handleSubmit }: FuelFormProps) {
+  /* Initializing state */
+  const [fuelData, setfuelData] = useState({
+    fuel_source_type: 'ng' as FuelSourceType,
+    fuel_source_unit: 'btu' as FuelSourceUnit,
+    fuel_source_value: 0 as number,
+  });
+
+  function submitForm(e: React.FormEvent) {
+    e.preventDefault();
+    console.log(fuelData);
+    handleSubmit(fuelData);
+  }
+
+  /* Function that updates the state */
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    /* Destructuring the event.target object */
+    const { name, value, valueAsNumber } = event?.target;
+    setfuelData((prevFuelData) => {
+      return {
+        ...prevFuelData,
+        [name]: name == 'fuel_source_value' ? valueAsNumber : value,
+      };
+    });
+  }
+
+  /* Renders the units conditionally */
+  const unitOptions =
+    fuelData.fuel_source_type == 'ng'
+      ? ['thousand_cubic_feet', 'btu']
+      : ['gallon', 'btu'];
+
+  function resetForm() {
+    setfuelData({
+      fuel_source_type: 'ng',
+      fuel_source_unit: 'btu',
+      fuel_source_value: 0,
+    });
+  }
+
+  return (
+    <div>
+      <Label htmlFor="fuel_source_type">Which fuel do you use?</Label>
+      <Select
+        id="fuel_source_type"
+        value={fuelData.fuel_source_type}
+        onChange={handleChange}
+        name="fuel_source_type"
+      >
+        <option value="">--- Choose</option>
+        <option value="ng">Natural Gas</option>
+        <option value="dfo">Home Heating and Diesel Fuel</option>
+        <option value="pg">Propane Gas</option>
+        <option value="ker">Kerosene</option>
+      </Select>
+
+      <br />
+
+      <Label htmlFor="unit">Which measuring unit do you use?</Label>
+      <Select
+        id="fuel_source_unit"
+        value={fuelData.fuel_source_unit}
+        onChange={handleChange}
```

Comment:
Writing your onChange functions like this works fine, but is vague. In the function declaration you state that you're expecting an argument, but when you read this line it's just handled automatically. I prefer to be explicit. `onChange={(event) => handleChange(event)}`